text/0033-view-hierarchy.md

```diff
@@ -0,0 +1,78 @@
+* Start Date: 2022-11-02
+* RFC Type: feature
+* RFC PR: [#33](https://github.com/getsentry/rfcs/pull/33)
+* RFC Status: -
+* RFC Driver: [Dhiogo Ramos Brustolin](https://github.com/brustolin)
+
+# Summary
+
+Allow SDKs to send a textual representation of the application's UI view hierarchy. 
+
+# Motivation
+
+Being able to see what the user was looking at is a good way to find problems in an application. This RFC proposes a way to send information from the UI state of an application, known as "view hierarchy". Compared to screenshots the view hierarchy provides complementary information about the UI, e.g. relationships between UI widgets or additional attributes not visible on the screen.
+
+Example View Hierarchies from PoC implementations:
+* [Android](https://github.com/getsentry/sentry-java/pull/1998#issuecomment-1104806065)
+* [iOS](https://github.com/getsentry/sentry-cocoa/pull/2044#issue-1329719543)
+* [Flutter](https://github.com/ueman/sentry-dart-tools/blob/b850db587f0e099bed253d13055c88c03d536875/sentry_flutter_plus/lib/src/integrations/tree_walker_integration.dart)
+* [Unity](https://github.com/getsentry/team-mobile/issues/64#issuecomment-1290868653)
+
+# Proposal
+
+Capture the view hierarchy in our front-end SDKs and convert it to a common JSON representation. Add it as an [attachment](https://develop.sentry.dev/sdk/envelopes/#attachment) to the envelope. The `attachment_type` is set to `"event.viewhierarchy"` and the `content_type` is set to the `"application/json"` mime-type.
```

Comment:
Do we want to use `event.view_hierarchy` instead of `event.viewhierarchy`? I'll let you make the final call on that.

Reply:
```suggestion
Capture the view hierarchy in our front-end SDKs and convert it to a common JSON representation. Add it as an [attachment](https://develop.sentry.dev/sdk/envelopes/#attachment) to the envelope. The `attachment_type` is set to `"event.view_hierarchy"` and the `content_type` is set to the `"application/json"` mime-type.
```

Reply:
I agree that `event.view_hierarchy` is more readable, but we have a precedent with `event.applecrashreport`, so I would stick to that convention: https://github.com/getsentry/relay/blob/36ca3675eae2202478cc7e001c2d8c9f189581c8/relay-server/src/envelope.rs#L334-L336

Reply:
I see. Thanks for the reference